snarkos/ledger/mod.rs

```diff
@@ -25,36 +37,81 @@ use std::{
     net::{IpAddr, SocketAddr},
     sync::Arc,
 };
-use tokio::task;
+use tokio::{
+    sync::{mpsc, oneshot},
+    task,
+};
 use warp::{reply, Filter, Rejection, Reply};
 
+/// Shorthand for the parent half of the `Ledger` message channel.
+pub(crate) type LedgerRouter<N> = mpsc::Sender<LedgerRequest<N>>;
+#[allow(unused)]
+/// Shorthand for the child half of the `Ledger` message channel.
+type LedgerHandler<N> = mpsc::Receiver<LedgerRequest<N>>;
+
+///
+/// An enum of requests that the `Ledger` struct processes.
+///
+#[derive(Debug)]
+pub enum LedgerRequest<N: Network> {
+    /// BlockResponse := (peer_ip, block)
+    BlockResponse(SocketAddr, Block<N>),
+    /// BlockRequest := (peer_ip, block_height)
+    BlockRequest(SocketAddr, u32),
+    /// Disconnect := (peer_ip, reason)
+    Disconnect(SocketAddr, DisconnectReason),
+    /// Failure := (peer_ip, failure)
+    Failure(SocketAddr, String),
+    /// Heartbeat
+    Heartbeat,
+    /// Ping := (peer_ip)
+    Ping(SocketAddr),
+    /// Pong := (peer_ip, block_height, node_type)
+    Pong(SocketAddr, u32, Option<NodeType>),
+    /// UnconfirmedBlock := (peer_ip, block)
+    UnconfirmedBlock(SocketAddr, Block<N>),
+    /// UnconfirmedTransaction := (peer_ip, transaction)
+    UnconfirmedTransaction(SocketAddr, Transaction<N>),
+}
```

Comment:
semantics: some of these variants are not _technically_ related to the `Ledger`, which is somewhat confusing; a `LedgerRequest` would ideally only contain `BlockRequest`, `BlockResponse`, `UnconfirmedBlock`, `UnconfirmedTransaction`